packages/bpk-component-navigation-bar/src/BpkNavigationBarButtonLink.tsx

```diff
@@ -42,18 +37,16 @@ const BpkNavigationBarButtonLink = ({
   className,
   ...rest
 }: Props) => (
-  <BpkButtonLink
-    // TODO: className to be removed
-    // eslint-disable-next-line @skyscanner/rules/forbid-component-props
-    className={getClassName(
-      'bpk-navigation-bar-button-link',
-      `bpk-navigation-bar-button-link--${barStyle}`,
-      className
-    )}
-    {...rest}
-  >
-    {children}
-  </BpkButtonLink>
+  <span className={className}>
+    <BpkButtonLink
+      alternate={barStyle === BAR_STYLES.onDark}
```

Comment:
Rather than setting our own class names we can use the BpkButtonLink's alterernate prop that flips the style.